File: 0019-remove-nth-node-from-end-of-list/0019-remove-nth-node-from-end-of-list.go
/**
 * Definition for singly-linked list.
 * type ListNode struct {
 *     Val int
 *     Next *ListNode
 * }
 */
func removeNthFromEnd(head *ListNode, n int) *ListNode {
  head = &ListNode{0, head}
  node, prev := head, head
  for node.Next != nil {
    if n > 0 {
      n--
    } else {
      prev = prev.Next
    }
    node = node.Next
  }
  prev.Next = prev.Next.Next
  return head.Next
}

File: 0019-remove-nth-node-from-end-of-list/0019-remove-nth-node-from-end-of-list.ts
/**
 * Definition for singly-linked list.
 * class ListNode {
 *     val: number
 *     next: ListNode | null
 *     constructor(val?: number, next?: ListNode | null) {
 *         this.val = (val===undefined ? 0 : val)
 *         this.next = (next===undefined ? null : next)
 *     }
 * }
 */

function removeNthFromEnd(head: ListNode | null, n: number): ListNode | null {
  head = new ListNode(0, head);
  let node = head, prev = head;
  while (node.next) {
    if (n-- <= 0) prev = prev.next;
    node = node.next;
  }
  prev.next = prev.next?.next;
  return head.next;
};